frontend/src/Views/AccountCreation/AccountForm.tsx

```diff
@@ -17,30 +18,23 @@ const useStyles = makeStyles((theme: Theme) => ({
 }));
 
 interface IProps {
-  onSubmit: (accountName: string, initialBalance: number) => void;
+  onFieldChange: (field: string, value: any) => void;
+  account: IAccount;
+  onSubmit?: (account: IAccount) => void;
 }
 
-const AccountForm = ({ onSubmit }: IProps) => {
+const AccountForm = ({ account, onFieldChange, onSubmit }: IProps) => {
   const classes = useStyles();
-
-  const [accountName, setAccountName] = React.useState("");
-  const [initialBalance, setInitialBalance] = React.useState("0");
-
-  const onAccountNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setAccountName(event.target.value);
-  };
-
-  const onInitialBalanceChange = (
-    event: React.ChangeEvent<HTMLInputElement>
-  ) => {
-    setInitialBalance(event.target.value);
+  const fieldUpdate = (fieldId: string) => (
+    e: ChangeEvent<HTMLInputElement>
+  ): void => {
+    onFieldChange(fieldId, e.target.value);
   };
-
-  const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
-    event.preventDefault();
-    onSubmit(accountName, +initialBalance);
+  const handleSubmit = (e: SyntheticEvent) => {
+    e.preventDefault();
+    // @ts-ignore
```

Comment:
Jeigu `IProps` butu palikus `onSubmit: ...` o ne `onSubmit?: ...` tai viskas veiktu ir nereiketu sito hacko

Reply:
aš kai kažką panašaus parašiau, tai pagal idėją tas `onSubmit` yra optional ir jo gali nebūt, bet čia keistas geriau ne ts-ignore uždėt o parašyt taip:
```typescript
    if (onSubmit) {
      onSubmit(...);
    }
```

Reply:
Kaip pagal ideja `AccountForm` gali neturetu onSubmit handlerio?

Reply:
jo šiaip jei state ne čia tai būtinai reik turėt onSubmit

Reply:
Hmm, gal is tikruju, jei state vietinis, galima neexposint `onSubmit`. Reik pagalvot/pasiskaitinet